akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ActiveActiveShardingExtension.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed
+
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Extension
+import akka.actor.typed.ExtensionId
+import akka.annotation.ApiMayChange
+import akka.annotation.DoNotInherit
+import akka.cluster.sharding.typed.internal.ActiveActiveShardingExtensionImpl
+import akka.cluster.sharding.typed.scaladsl.EntityRef
+import akka.persistence.typed.ReplicaId
+import java.util.{Map => JMap}
+
+/**
+ * Extension for running active active in sharding by starting one separate instance of sharding per replica.
+ * The sharding instances can be confined to datacenters or cluster roles or run on the same set of cluster nodes.
+ */
+@ApiMayChange
+object ActiveActiveShardingExtension extends ExtensionId[ActiveActiveShardingExtension] {
+
+  override def createExtension(system: ActorSystem[_]): ActiveActiveShardingExtension =
+    new ActiveActiveShardingExtensionImpl(system)
+
+  def get(system: ActorSystem[_]): ActiveActiveShardingExtension = apply(system)
+
+}
+
+/**
+ * Not for user extension.
+ */
+@DoNotInherit
+@ApiMayChange
+trait ActiveActiveShardingExtension extends Extension {
```

Comment:
A ActiveActive is always about `EventSourcedBehavior` but the API at sharding level is about `EntityRef`. 

I don't know if there is anything we can do about it, but it can give the impression that this can be used with any sort of `EntityRef`.

Reply:
I tried doing a safety net by replacing the behavior factory from the entity with a wrapper checking the behavior but that was only enough for Java and not really possible for Scala without a lot of changes so backed out of it.